src/components/tile/tile.test.tsx

```diff
@@ -0,0 +1,184 @@
+  expect(styledTile).toHaveAttribute("data-role", "bar");
+});
+
+test.each(TILE_HIGHLIGHT_VARIANTS)(
+  "should render with the highlight element when `highlightVariant` is passed %s",
+  (highlightVariant) => {
+    render(<Tile highlightVariant={highlightVariant} />);
+    const highlightElement = screen.getByTestId(
+      `tile-${highlightVariant}-highlight`
+    );
+
+    expect(highlightElement).toBeVisible();
+    expect(highlightElement).toHaveAttribute("aria-hidden");
+  }
+);
+
+/* tests for coverage */
+test.each<[TileProps["borderVariant"], string]>([
+  ["selected", "colorsUtilityYin100"],
+  ["positive", "colorsSemanticPositive500"],
+  ["negative", "colorsSemanticNegative500"],
+  ["caution", "colorsSemanticCaution500"],
+  ["info", "colorsSemanticInfo500"],
+])(
+  "renders with expected border when borderVariant is set to %s",
+  (borderVariant, borderVariantToken) => {
+    render(
+      <Tile borderVariant={borderVariant} data-role="tile">
+        <TileContent>Child 1</TileContent>
+        <TileContent>Child 2</TileContent>
+      </Tile>
+    );
+
+    const tileElement = screen.getByTestId("tile");
+
+    expect(tileElement).toHaveStyleRule(
+      "border",
+      `var(--borderWidth100) solid var(--${borderVariantToken})`
+    );
+  }
+);
+
+test('renders with expected background and border styles when variant is "active"', () => {
+  render(
+    <Tile variant="active" data-role="tile">
+      <TileContent>Child 1</TileContent>
+      <TileContent>Child 2</TileContent>
+    </Tile>
+  );
+
+  const tileElement = screen.getByTestId("tile");
+
+  expect(tileElement).toHaveStyle({
+    backgroundColor: "var(--colorsActionMajor025)",
+    border: "var(--borderWidth100) solid var(--colorsActionMajor500)",
+  });
+});
+
+test('renders with expected background and border styles when variant is "grey"', () => {
+  render(
+    <Tile variant="grey" data-role="tile">
+      <TileContent>Child 1</TileContent>
+      <TileContent>Child 2</TileContent>
+    </Tile>
+  );
+
+  const tileElement = screen.getByTestId("tile");
+
+  expect(tileElement).toHaveStyle({
+    backgroundColor: "var(--colorsUtilityMajor025)",
+    border: "var(--borderWidth100) solid var(--colorsUtilityMajor200)",
+  });
+});
+
+test.each<TileProps["roundness"]>(["default", "large", "small"])(
+  "renders with the expected border radius when roundness is %s",
+  (roundness) => {
+    render(
+      <Tile roundness={roundness} data-role="tile">
+        <TileContent>Child 1</TileContent>
+        <TileContent>Child 2</TileContent>
+      </Tile>
+    );
+
+    const tileElement = screen.getByTestId("tile");
+    let expectedBorderRadius: string;
+
+    switch (roundness) {
+      case "small":
+        expectedBorderRadius = "var(--borderRadius050)";
+        break;
+      case "large":
+        expectedBorderRadius = "var(--borderRadius200)";
+        break;
+      default:
+        expectedBorderRadius = "var(--borderRadius100)";
+        break;
+    }
```

Comment:
**suggestion**: it might be worth considering introducing an `expectedBorderRadius` parameter for the tests. This could help simplify the test logic and improve readability by removing the conditional logic. What do you think?

```tsx
test.each([
  ["default", "var(--borderRadius050)"],
  ["large", "var(--borderRadius200)"],
  ["small", "var(--borderRadius100)"],
] as const)(
  "renders with the expected border radius when roundness is %s",
  (roundness, expectedBorderRadius) => {
    render(
      <Tile roundness={roundness} data-role="tile">
        <TileContent>Child 1</TileContent>
        <TileContent>Child 2</TileContent>
      </Tile>
    );

    const tileElement = screen.getByTestId("tile");
    expect(tileElement).toHaveStyleRule("border-radius", expectedBorderRadius);
  }
);

```